pennylane_lightning_gpu/src/util/CMakeLists.txt

```
@@ -1,6 +1,39 @@
-project(lightning_gpu_utils LANGUAGES CXX CUDA)
-set(CMAKE_CXX_STANDARD 20)
+project(lightning_gpu_utils LANGUAGES CUDA CXX)
+enable_language(CUDA)
+set(CMAKE_CXX_STANDARD 17)
+
+find_package(CUDAToolkit REQUIRED)
+
+include(CheckLanguage)
+check_language(CUDA)
+
+if(NOT DEFINED CMAKE_CUDA_STANDARD)
+    set(CMAKE_CUDA_STANDARD 17)
+    set(CMAKE_CUDA_STANDARD_REQUIRED ON)
+endif()
+
+
+set(DATA_BUFFER_FILES DataBuffer.cu CACHE INTERNAL "" FORCE)
+
+add_library(lightning_gpu_utils STATIC ${DATA_BUFFER_FILES})
+
+set_target_properties( lightning_gpu_utils
+                       PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
+
+set_target_properties( lightning_gpu_utils
+                       PROPERTIES CUDA_RESOLVE_DEVICE_SYMBOLS ON)
+
+set_target_properties(lightning_gpu_utils PROPERTIES CUDA_ARCHITECTURES "80;86")
```

```suggestion
set_target_properties(lightning_gpu_utils PROPERTIES CUDA_ARCHITECTURES "80;86")
```
We need to support the SM7.0 and all above architectures here too. Can we set a minimum arch set, rather than being explicit?

---

What about set `CUDA_ARCHITECTURES` as `native`? This option will build target for the (concrete) CUDA architectures of GPUs available at configuration time.

---

Well, `native` is not a good option for the CI since there is no gpu.